### PR TITLE
Standardize clone depth of drawable configurations

### LINKED ISSUE #987: Review and standardize clone depth of drawable configurations
### Background

Issue #964 replaces `ICloneable` with strongly typed `Clone()` methods while intentionally preserving the existing cloning behavior.

During that refactoring, the drawable configuration hierarchy shows inconsistent cloning depth. Some mutable nested data is copied, while other nested objects and collections are retained by reference.

Examples:

* `DrawableConfig.Clone()` creates new lists using `ToList()`, but the contained `LineConf`, `TextConf`, `ImageConf`, and `MindMapNodeConf` instances are still shared with the original configuration.
* `MindMapNodeConf.Clone()` retains references to mutable nested objects and collections such as `Children`, `childrenNames`, `BorderConf`, and `TextConf`.
* `BranchLineConf` in `MindMapNodeConf` should also be reviewed because it is currently not copied by the clone implementation.
* `LineConf.Clone()` already creates a new renderer position array and clones its line cap configurations.
* `ImageConf.Clone()` creates a new `FileData` array.
* `TextConf` and `LineCapConf` currently contain only value-like or immutable data and therefore do not require additional recursive copying.

### Goal

Define and document the intended cloning semantics for drawable configurations and make mutable nested state independent where that is required by callers.

### Tasks

1. Review every `Clone()` method in the drawable configuration hierarchy and explicitly decide which nested members must be copied and which may intentionally remain shared.
2. Determine whether `DrawableConfig.Clone()` should clone the configuration objects contained in its lists instead of only copying the lists themselves.
3. Review `MindMapNodeConf.Clone()` in particular, including `Children`, `childrenNames`, `BorderConf`, `TextConf`, and `BranchLineConf`.
4. Preserve intentional sharing where appropriate and document it explicitly.
5. Add regression tests that verify the chosen ownership semantics by mutating cloned collections and nested configuration objects and checking whether the original is affected as intended.
6. Update XML documentation so that each `Clone()` method clearly states whether mutable nested state is copied or shared.

### PR DESCRIPTION
**Pull request by Cyclone1337**
Closes #987

## Summary

This PR reviews and standardizes the clone depth of drawable configurations.

The contained configurations of `DrawableConfig` are now cloned independently instead of being retained by reference. Mutable nested state in the individual drawable configurations is cloned where appropriate, while intentional sharing is preserved and documented.

## Changes

* Deep clone the configurations contained in `DrawableConfig`.
* Deep clone nested configurations and collections in `MindMapNodeConf`, including `BranchLineConf`.
* Preserve the intentional sharing of `GameObject` references contained in `MindMapNodeConf.Children`.
* Preserve null `ImageConf.FileData` values while independently copying existing file data.
* Validate required image data before creating `AddImageNetAction`.
* Add `CloneWithoutDrawableTypes()` to avoid unnecessary deep cloning in surface network actions.
* Document the clone depth of all relevant drawable configuration types.
* Add regression tests for mutable collections, nested configurations, null values, and clone independence.

## Testing

* Added and executed regression tests for all relevant drawable configuration clone methods.
* Executed the complete EditMode test suite successfully.
* Verified the clone behavior at runtime in SEE, including line caps, image file data, mind map branch configurations, `DrawableConfig.Clone()`, and `CloneWithoutDrawableTypes()`.

### CODE REVIEW

File: Assets/SEE/Net/Actions/Drawable/AddImageNetAction.cs [hunk 1]
```diff
@@ -51,6 +58,29 @@ public AddImageNetAction(string drawableID, string parentDrawableID, ImageConf i
...
+        /// Thrown if <paramref name="imageConf"/> does not contain file data.
+        /// </exception>
+        internal static void ValidateImageConfiguration(ImageConf imageConf)
+        {
+            if (imageConf == null)
+            {
+                throw new ArgumentNullException(nameof(imageConf));
+            }
+
+            if (imageConf.FileData == null)
```
[the remark this answers is not in the document]
Comment (Cyclone1337) [reply Assets/SEE/Net/Actions/Drawable/AddImageNetAction.cs]: Thanks for pointing this out. I updated the validation to reject zero-length file data in addition to null values and added a regression test for this case. The web image case remains unaffected because the file data is cleared only after the initial configuration has been validated.

File: Assets/SEETests/Game/Drawable/Configurations/DrawableConfigCloneTests.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace SEE.Game.Drawable.Configurations
+{
+    /// <summary>
+    /// Tests the cloning semantics of <see cref="DrawableConfig"/>.
+    /// </summary>
+    [TestFixture]
+    public class DrawableConfigCloneTests
```
Comment (koschke) [inline Assets/SEETests/Game/Drawable/Configurations/DrawableConfigCloneTests.cs]: I renamed it to `TestDrawableConfigClone`. Our tests should start with `Test` uniformly.

**Review by koschke — COMMENTED** [review]
There were only very minor things, which I fixed myself.
I renamed the new test classes so that they start with the prefix Test for consistency with our other test classes. 

I will ask Claude for a second opinion tomorrow.

**Comment by koschke** [comment]
Here is Claude's opinion.

## Review: `improvement/987-standardize-drawable-clone-depth`

18 commits, 28 files, +739/−50 vs `master`. Eight production files (six drawable config classes, two net actions); the rest is new tests plus `.meta` files. The change replaces three `TODO (#987)` markers with actual clone semantics.

### Findings

- [[DrawableConfig.cs:183](https://claude.ai/epitaxy/Assets/SEE/Game/Drawable/Configurations/DrawableConfig.cs:183)](Assets/SEE/Game/Drawable/Configurations/DrawableConfig.cs:183) — `Clone()` is the only clone method in this PR left without null handling for its collections, while the same class guards those lists in `SaveAttributes`.

That is the only issue I'd raise, and it is a consistency gap rather than a regression — the previous `ToList()` threw on null too.

### What I checked and found correct

Several things in this diff look risky and turn out to be fine; worth recording so they don't get re-litigated:

- **`SurfaceNetAction` behavior is preserved exactly.** The old code cloned then called `.Clear()` on four lists; the new `CloneWithoutDrawableTypes()` omits them entirely. That is equivalent only because all four fields carry `= new()` initializers, so the omitted properties come back as empty lists rather than null. I verified the initializers rather than assuming them — had they been absent, this would have silently turned empty lists into nulls across every surface net action.
- **The new `AddImageNetAction` validation does not break the URL-image path**, which was my main suspicion, since the constructor demands non-empty `FileData` and then deliberately empties it for web images. Every producer supplies real data: `ImageConf.GetImageConf` reads from `ImageValueHolder`, `AddImageAction` captures its memento straight after `PlaceImage`, and on the receiving client both web branches refill `FileData` before `RePlaceImage`. The one path that yields null — `ImageConf.Restore` when the file is missing locally — already threw `NullReferenceException` inside `Clone()` before this change, so the new `ArgumentException` is strictly more informative.
- **`MindMapNodeConf` now cloning `BranchLineConf` is a fix, not a behavior break.** Previously the field was absent from the object initializer, so clones silently defaulted it to null. The consumer at `GameMindMap.cs:969` is null-guarded, and the practical beneficiary is `DrawableConfigManager.GetSinglePageConfig`, which previously lost branch-line styling when extracting a page.
- **The tests can see the `internal` members** they call (`ValidateImageConfiguration`, `CloneWithoutDrawableTypes`). `SEETests` is a separate assembly, but `Assets/SEE/SEE.cs` declares `[assembly: InternalsVisibleTo("SEETests")]`.
- **The branch compiles.** Roslyn against `Library/ScriptAssemblies` gives the same 1644-error baseline as `master`, with no errors in any of the eight changed production files.

### Two notes outside the diff

`ImageConf.Restore` reads the URL from `pathLabel` instead of `urlLabel` ([[ImageConf.cs:157](https://claude.ai/epitaxy/Assets/SEE/Game/Drawable/Configurations/ImageConf.cs:157)](Assets/SEE/Game/Drawable/Configurations/ImageConf.cs:157)), so every restored image gets `URL == Path`. That matters here because it makes `!string.IsNullOrEmpty(Conf.URL)` true for restored *local* images, sending them down `AddImageNetAction`'s web branch, where the receiver will try to download a filesystem path. Pre-existing and untouched by this PR, but it sits directly under the code being changed.

The `childrenNames != null` guard added to `MindMapNodeConf.Clone()` is unreachable — that field has an initializer and is never assigned null anywhere in the class. Harmless, just not load-bearing like the `Children` guard beside it.

Finally, `TestAddImageNetAction` covers the empty and non-empty `FileData` cases but not the documented `ArgumentNullException` for a null `imageConf`. Worth adding, and note NUnit's `Assert.Throws<T>` matches the exact type, so `Assert.Throws<ArgumentException>` would *not* accept an `ArgumentNullException` if the validation were ever reshuffled.

**Comment by Cyclone1337** [comment]
Thanks for getting Claude's second opinion. I addressed all of the points raised in the review.

`DrawableConfig.Clone()` now preserves null configuration lists and has a corresponding regression test. I also added the missing test for the documented `ArgumentNullException` in `AddImageNetAction`.

I additionally fixed the pre-existing `ImageConf.Restore()` issue found by Claude, so the URL is now restored from `urlLabel` instead of `pathLabel`, and added a regression test for it. Finally, I simplified the redundant null handling for `childrenNames` in `MindMapNodeConf.Clone()`.

**Comment by koschke** [comment]
So, should I pull the merge trigger?

**Comment by Cyclone1337** [comment]
Yes, from my side everything is ready now. Thanks!
